### PR TITLE
Added DB week2 homework

### PR DESCRIPTION
**Pull request by sandhana14**

### CODE REVIEW

File: databases/week2/DB_week2.sql [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+-- Change a task status
+UPDATE task set status_id = 3 WHERE id=1;
+-- Mark a task as complete
+UPDATE task set status_id = 3 WHERE task_title="Charge electric bicycle" AND id=22;
+-- Delete a task
+DELETE FROM task where id=22;
+
+-- Part 2: School database
+-- Creating class table
+CREATE TABLE class(id INT PRIMARY KEY,name VARCHAR(50),begins DATE,ends DATE);
```
Comment (aina21) [inline databases/week2/DB_week2.sql:19]: It's better to add AUTO_INCREMENT for primary keys :)

File: databases/week2/DB_week2.sql [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+class_id INT,FOREIGN KEY (class_id) REFERENCES class(id));
+-- Create an index on the name column of the student table.
+CREATE INDEX idx_name ON student (name);
+-- Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished
+ALTER TABLE class ADD status ENUM("not-started", "ongoing", "finished");
+
+-- Part 3: More queries
+-- Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT task.* FROM task 
+JOIN user_task ON task.id=user_task.task_id
```
Comment (aina21) [inline databases/week2/DB_week2.sql:31]: I believe we don't have a user_task table, you can join task table to the user as it has user_id.

File: databases/week2/DB_week2.sql [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+JOIN user ON user.id= user_task.user_id
+WHERE user.email LIKE "%@spotify.com";
+
+-- Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT task.* FROM task 
+JOIN user_task ON task.id=user_task.task_id
```
Comment (aina21) [inline databases/week2/DB_week2.sql:37]: same here!
try without user_task, and the result should be zero :)

File: databases/week2/DB_week2.sql [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+JOIN user ON user.id= user_task.user_id
+JOIN status ON task.status_id = status.id
+WHERE user.name = "Donald Duck" AND status.name = "Not started";
+
+-- Get all the tasks for 'Maryrose Meadows' that were created in september
+SELECT task.* FROM task 
+JOIN user_task ON task.id = user_task.task_id
```
Comment (aina21) [inline databases/week2/DB_week2.sql:44]: same

**Review by aina21 — CHANGES_REQUESTED** [review]
good job :)

**Review by aina21 — APPROVED** [review]

File: databases/week2/DB_week2.sql [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                       due_date DATE NULL);
+                       
+CREATE TABLE employees_projects(id INT NOT NULL AUTO_INCREMENT,
+                                employee_id INT NOT NULL,
+                                project_id INT NOT NULL,
+                                FOREIGN KEY (employee_id) REFERENCES employees(id) ON DELETE CASCADE,
+                                FOREIGN KEY (project_id) REFERENCES projects(id) ON DELETE CASCADE,
+                                PRIMARY KEY (id));
+
+
```
Comment (soujanyapolavarapu) [inline databases/week2/DB_week2.sql:72]: good job sandhana :) I have reviewed your queries...looks clean and prefect.

**Review by soujanyapolavarapu — COMMENTED** [review]
good job dear :)

**Pull request merged**